### DISCUSSION TITLE
Test fails

### DISCUSSION THREAD

**Discussion by abetomo** [groonga/groonga#1659] [Q&A] [answered]
If you run the test with the Dockerfile shown below, the test will fail.

```Dockerfile
FROM ubuntu:22.04

WORKDIR /build

RUN apt-get update && apt-get -y upgrade && apt-get -y install git ruby ruby-dev
RUN git clone --recursive https://github.com/groonga/groonga.git
RUN cd groonga && ./setup.sh
RUN cmake -B ./groonga.build -S ./groonga -GNinja -DCMAKE_BUILD_TYPE=Debug
RUN ninja -C ./groonga.build
ENV BUILD_DIR=/build/groonga.build/test/command

CMD ["./groonga/test/command/run-test.sh"]
```

The log of the test is attached. [run-test.log](https://github.com/groonga/groonga/files/13881665/run-test.log)
The test stopped, so I terminated with `kill`.

The difference in timezone seemed to be the cause.
However, I could not solve the problem and would be grateful for any hints on how to solve it.

**Comment by kou** [comment]
I agree that one of these failures is related to timezone.
We assume `TZ=Asia/Tokyo` timezone and `test/command/run-test.sh` sets it explicitly.
Could you check whether `tzdata` is installed or not?

**Comment by abetomo** [comment]
Thanks for the answer.
Some tests passed with `tzdata` installation.
I'm attaching the logs as there are still some tests that fail.  
[run-test.log](https://github.com/groonga/groonga/files/13881818/run-test.log)

**Comment by kou** [comment]
Could you install `libmecab-dev` and `mecab-naist-jdic`?
Some tests use MeCab.

**Reply by kou** [reply]
It may be better that our tests that use MeCab are omitted automatically when MeCab isn't supported.

**Reply by abetomo** [reply]
Thanks. I did the install and passed the test.

**Comment by kou** [comment]
It seems that geo related tests are timed out.
In general, Groonga is crashed with backtrace for these cases. But backtrace wasn't shown. Is it related to Docker?

**Comment by kou** [comment]
```text
+#@copy-path #{db_path}.0000101 #{db_path}.0000101.wal#|e| [copy-path] failed to copy: Errno::ENOSPC: No space left on device - copy_file_range
```

"No space left on device"?

**Reply by abetomo** [reply]
In the docker container, `/dev/shm` was only 64MB.
I increased the size with `--shm-size=32g` and it passed.
(`docker container run --shm-size=32g ...`)

This also solved the geo failure.

**Comment by abetomo** [comment, marked as answer]
* Install `tzdata`
* Install `libmecab-dev` and `mecab-naist-jdic`
* Run with `--shm-size=32g` option
    * Example: `docker container run --shm-size=32g ...`
